mitreattack/navlayers/layerGenerator_cli.py

```diff
@@ -10,35 +10,30 @@
 
 def main():
     parser = argparse.ArgumentParser(description='Generate an ATT&CK Navigator layer')
-    parser.add_argument('-m', '--mode', choices=['overview', 'usage'], required=True,
-                        help='Generate a complete overview or the specific usage for a entity')
-    parser.add_argument('-t', '--type', choices=['group', 'software', 'mitigation'], required=True,
-                        help='What kind of object to build the layer for')
-    parser.add_argument('-i', '--identity', help='The identity (id/name) of the desired entity '
-                                                 '(required for usage generation)', default=None)
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--overview-type', choices=['group', 'software', 'mitigation'],
+                       help='Output a matrix where the target type is summarized across the entire dataset.')
+    group.add_argument('--mapped-to', help='Output techniques mapped to the given group, software, or mitigation. '
+                                           'Argument can be name, associated group/software, or ATT&CK ID.')
     parser.add_argument('-o', '--output', help='Path to the output layer file', default='generated_layer.json')
-    parser.add_argument('--matrix', help='Which matrix to build off of', default='enterprise')
+    parser.add_argument('--domain', help='Which domain to build off of', default='enterprise')
```

Comment:
May be worth adding the `choices` argument to `--domain` to make sure they don't put in a bogus domain argument.

Reply:
Less "shooting oneself in the foot" is always a good idea.